### PR TITLE
refactor: Improve performance

### PR DESCRIPTION
**Pull request by howjmay**
For non-primitive types, using prefix ++/-- operator is faster.

And two not necessary if condition are removed.

### CODE REVIEW

File: urdf_parser/src/check_urdf.cpp [hunk 1]
```diff
@@ -44,7 +44,7 @@ void printTree(LinkConstSharedPtr link,int level = 0)
 {
   level+=2;
   int count = 0;
-  for (std::vector<LinkSharedPtr>::const_iterator child = link->child_links.begin(); child != link->child_links.end(); child++)
+  for (std::vector<LinkSharedPtr>::const_iterator child = link->child_links.begin(); child != link->child_links.end(); --child)
```
Comment (clalancette) [inline urdf_parser/src/check_urdf.cpp:47]:
```suggestion
  for (std::vector<LinkSharedPtr>::const_iterator child = link->child_links.begin(); child != link->child_links.end(); ++child)
```

**Review by clalancette — CHANGES_REQUESTED** [review]

File: urdf_parser/src/check_urdf.cpp (hunk 1 above)
Comment (howjmay) [reply urdf_parser/src/check_urdf.cpp]: fixed

**Review by clalancette — APPROVED** [review]

**Comment by clalancette** [comment]
CI:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/17376/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/11910/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/17878/)

**Review by rhaschke — APPROVED** [review]
@howjmay, your changes do not harm, but:

"On any decent compiler, ++i and i++ are identical if the value is not used."

https://stackoverflow.com/questions/2020184/preincrement-faster-than-postincrement-in-c-true-if-yes-why-is-it

**Pull request merged**